soh/src/overlays/actors/ovl_En_Heishi1/z_en_heishi1.c

```diff
@@ -112,14 +112,25 @@ void EnHeishi1_Init(Actor* thisx, GlobalContext* globalCtx) {
         }
     }
 
+    // eventChkInf[4] & 1 = Got Zelda's Letter
+    // eventChkInf[5] & 0x200 = Got item from impa
+    // eventChkInf[8] & 1 = Ocarina thrown in moat
+    
+    // Rando'd but not met Zelda yet
+    bool randoNotMetZelda =
+        (gSaveContext.n64ddFlag) && !(gSaveContext.eventChkInf[5] & 0x200) && !(gSaveContext.eventChkInf[4] & 1);
+    // Rando'd and already met Zelda
+    bool randoMetZelda = (gSaveContext.n64ddFlag) && (gSaveContext.eventChkInf[5] & 0x200) && (gSaveContext.eventChkInf[4] & 1);
```

Comment:
not sure how i feel about these

i get the desire to clean up the if statements, but i think combining the rando check and the met zelda check may be going a little too far

i also don't like there being 2 of them, i think ideally we'd have

```
if ((!rando && non-rando-logic) || (rando && !metZelda)) {
```
and
```
if((!rando && non-rando-logic) || (rando && metZelda)) {
```

Reply:
I went back and forth quite a few times on how to handle this myself. I'll change it to what you suggested though, looks better than what I was doing.